### DISCUSSION TITLE
Refactor Highlights to Filter-like system

### DISCUSSION THREAD

**Discussion by Infinitay** [Chatterino/chatterino2#3710] [Ideas]
Currently we have two different systems between highlights and filters. Highlights currently allow us to highlight according to the message content. Filters on the other hand allow us to do so much more and filter messages by a [multitude of variables](https://wiki.chatterino.com/Filters/#variables) such as channel, message, or user information. Additionally, users can highlight certain usernames or badges, and also allow for blacklisting usernames from being highlighted.

Changing our current highlights to using the same logic available to filters would not only make things more uniform, but also allow for more options in highlighting messages. The current highlight options are miniscule compared to what is offered by filters.

Some issues related to this:
#3123, #2365, #2354, #3705 (assuming additional filter support for this), #3616, #3323

**Comment by pajlada** [comment]
https://github.com/Chatterino/chatterino2/pull/3399 works to rebuiuld the highlight system into a standalone testable system

**Reply by Infinitay** [reply]
I don't know how I avoided that when I looked for other issues. Any idea on how backlogged this is?

**Reply by pajlada** [reply]
The PR has been made up-to-date with the latest commit now, I think some further testing is all that was required for it. Started adding some unit tests & benchmarks

**Comment by Mm2PL** [comment]
I think this could result in hard to use ui. Also if we were to do this, ignores should probably follow.

**Reply by Infinitay** [reply]
It would have the same ui as current Filters. If you mean because the current Highlights page would be _cramped_ because of potentially long highlights, perhaps we could allow for vertical growth for the entries instead of having them fixed?
